pkg/endpoint/id/id.go

```diff
@@ -67,20 +67,13 @@ func ParseCiliumID(id string) (int64, error) {
 	return n, nil
 }
 
-// FIXME:
-//  - Add docker ID and docker endpoint parsers
-
-// ParseID parses specified id and returns normalized id as string.
-func ParseID(id string) (PrefixType, string, error) {
+// ParsePrefix parses the PrefixType from the given id that contains the prefix
+// and an ID. Returns a PrefixType and the ID. An error is returned if the prefix
+// is not known by the constants of valid prefixes.
+func ParsePrefix(id string) (PrefixType, string, error) {
 	prefix, eid := SplitID(id)
 	switch prefix {
-	case CiliumLocalIdPrefix:
-		if _, err := ParseCiliumID(id); err != nil {
```

Comment:
Is it safe to remove this validation?

Reply:
Yes, there are only 2 callers of that function, of them doesn't even use the returned `eid`, the other executes ParseCiliumID afterwards